index.html

```diff
@@ -1692,71 +1692,52 @@ <h3>Add Proof</h3>
           section <a href="#add-proof"></a>.
-          A <dfn>secured data document</dfn> is produced as
-          output. Whenever this algorithm encodes strings, it MUST use UTF-8 encoding.
+          A new <dfn>secured data document</dfn> is produced as
+          output. Whenever this algorithm encodes strings, it MUST use UTF-8 
+          encoding.
         </p>
 
         <ol class="algorithm">
           <li>
-Let <var>output</var> be a copy of <var>unsecuredDocument</var> and 
-<var>proof_list</var> be an empty list.
+Let <var>proof</var> be set to <var>securedDocument</var>.<var>proof</var>. Let 
+<var>proof_list</var> be an empty list. If <var>proof</var> is a list copy all 
+the elements of <var>proof</var> to <var>proof_list</var>. If <var>proof</var> 
+is an object add a copy of that object to <var>proof_list</var>.
           </li>
           <li>
-For each <var>options</var> in <var>options_list</var> run steps 2 through 8 of
-the algorithm in section <a href="#add-proof"></a>. If no exceptions are raised 
-append the generated <var>proof</var> value to the <var>proof_list</var> 
-otherwise raise the exception.
+Let the <var>unsecuredDocument</var> be a copy of the <var>securedDocument</var> 
+with the <var>proof</var> attribute removed. Let <var>output</var> be a copy of 
+the <var>unsecuredDocument</var>.
           </li>
           <li>
-Set <var>output</var>.<var>proof</var> to the value of <var>proof_list</var>.
-          </li>
-          <li>
-Return <var>output</var> as the <a>secured data document</a>.
+If <var>options</var> contains a <code>previousProof</code> attribute check if 
+an element of the <var>proof_list</var> has a matching <code>id</code> attribute. 
+If not  a `PROOF_GENERATION_ERROR` MUST be raised.
           </li>
-        </ol>
-      </section>
-      <section>
-        <h3>Add Proof Chain</h3>
-        <p>
-          The following algorithm specifies how to add a proof chain to a 
-          document. Required inputs are an
-          <a>unsecured data document</a> (<var>unsecuredDocument</var>) and a
-          (unordered) list of
-          <a>proof options</a> (<var>options_list</var>) to which an order will
-          be established. Each <a>proof options</a>
-          (<var>options</var>) in the list must satisfy the criteria in 
-          section <a href="#add-proof"></a>.
-          A <dfn>secured data document</dfn> is produced as
-          output. Whenever this algorithm encodes strings, it MUST use UTF-8 encoding.
-        </p>
-
-        <ol class="algorithm">
           <li>
-For each <var>options</var> in <var>options_list</var> that represents a proof 
-that must come before another proof add an <code>id</code> attribute if it is 
-not already present.
+Run steps 2 through 8 of the algorithm in section <a href="#add-proof"></a>. If 
+no exceptions are raised append the generated <var>proof</var> value to the 
+<var>proof_list</var> otherwise raise the exception.
```

Comment:
```suggestion
no exceptions are raised, append the generated <var>proof</var> value to the 
<var>proof_list</var>; otherwise, raise the exception.
```